src/main/java/com/descope/proxy/impl/AbstractProxyImpl.java

```diff
@@ -104,15 +104,19 @@ public R handleResponse(ClassicHttpResponse response) throws HttpException, IOEx
                   bs.toString(), String.valueOf(res.getCode()), bs.toString());
               }
             }
-            R r = returnClz != null
-                ? objectMapper.readValue(tee, returnClz)
-                : objectMapper.readValue(tee, typeReference);
-            if (log.isDebugEnabled()) {
-              String resStr = bs.toString();
-              log.debug(String.format("Received response %s",
-                  resStr.substring(0, resStr.length() > 10000 ? 10000 : resStr.length())));
+            try {
```

Comment:
Why is this suddenly needed ?

Reply:
Cleaner handling of parsing error - as we promise we throw our own exception and not others